src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.h

```diff
@@ -155,49 +215,72 @@ struct BallPlacementPlayFSM
-            KickOffWallState_S + Update_E[!kickDone_G && shouldKickOffWall_G] /
-                                     kickOffWall_A                = KickOffWallState_S,
-            KickOffWallState_S + Update_E[kickDone_G]             = KickOffWallState_S,
-            KickOffWallState_S + Update_E[!kickDone_G]            = AlignPlacementState_S,
-            AlignPlacementState_S + Update_E[shouldKickOffWall_G] = KickOffWallState_S,
+            StartState_S + Update_E[shouldPickOffWall_G] = AlignWallState_S,
+            AlignWallState_S + Update_E[!wallAlignDone_G && shouldPickOffWall_G] /
+                                   alignWall_A = AlignWallState_S,
+            AlignWallState_S + Update_E[wallAlignDone_G] / setPickOffDest_A =
+                PickOffWallState_S,
+            AlignWallState_S + Update_E[!shouldPickOffWall_G] = AlignPlacementState_S,
+            PickOffWallState_S + Update_E[!wallPickOffDone_G] / pickOffWall_A =
+                PickOffWallState_S,
+            PickOffWallState_S + Update_E[wallPickOffDone_G] / startWait_A =
+                PickOffWaitState_S,
+            PickOffWaitState_S + Update_E[!waitDone_G] / wait_A = PickOffWaitState_S,
+            PickOffWaitState_S + Update_E[waitDone_G]           = ReleaseWallState_S,
+            ReleaseWallState_S + Update_E[!ballReleased_G] / releaseBall_A =
+                ReleaseWallState_S,
+            ReleaseWallState_S + Update_E[ballReleased_G]         = AlignPlacementState_S,
+            AlignPlacementState_S + Update_E[shouldPickOffWall_G] = AlignWallState_S,
             AlignPlacementState_S + Update_E[!alignDone_G] / alignPlacement_A =
                 AlignPlacementState_S,
             AlignPlacementState_S + Update_E[alignDone_G]            = PlaceBallState_S,
             PlaceBallState_S + Update_E[!ballPlaced_G] / placeBall_A = PlaceBallState_S,
             PlaceBallState_S + Update_E[ballPlaced_G] / startWait_A  = WaitState_S,
-            WaitState_S + Update_E[!waitDone_G]                      = WaitState_S,
-            WaitState_S + Update_E[waitDone_G]                       = RetreatState_S,
-            RetreatState_S + Update_E[retreatDone_G && ballPlaced_G] = X,
-            RetreatState_S + Update_E[ballPlaced_G] / retreat_A      = RetreatState_S);
+            WaitState_S + Update_E[!waitDone_G && ballPlaced_G] / wait_A = WaitState_S,
+            WaitState_S + Update_E[!ballPlaced_G]                        = StartState_S,
+            WaitState_S + Update_E[waitDone_G]                           = RetreatState_S,
+            RetreatState_S + Update_E[retreatDone_G && ballPlaced_G]     = X,
+            RetreatState_S + Update_E[ballPlaced_G] / retreat_A          = RetreatState_S,
+            RetreatState_S + Update_E[!ballPlaced_G]                     = StartState_S);
     }
 
    private:
     TbotsProto::AiConfig ai_config;
-    std::shared_ptr<WallKickoffTactic> pivot_kick_tactic;
+    std::shared_ptr<MoveTactic> align_wall_tactic;
```

Comment:
this needs to be a `PlaceBallTactic`. Right now, `MoveTactic`s during ball placement will be unable to enter the stadium obstacle and we wont be able to align correctly (see `ball_placement_play_test`). `PlaceBallTactic` is a copy of `MoveTactic` that removes this obstacle (https://github.com/UBC-Thunderbots/Software/blob/dea10f841ad24ef8ac8da743416e4c9495f9e83b/src/software/ai/hl/stp/tactic/dribble/dribble_tactic.h#L52). We also have an equivalent `PlaceBallDribbleTactic`. I would suggest renaming these tactics to `BallPlacementMoveTactic` and `BallPlacementDribbleTactic`.